lib/wrappers/openssl.nim

```diff
@@ -189,16 +210,41 @@ const
 
 proc TLSv1_method*(): PSSL_METHOD{.cdecl, dynlib: DLLSSLName, importc.}
 
+# TLS_method(), TLS_server_method(), TLS_client_method() are introduced in 1.1.0
+# and support SSLv3, TLSv1, TLSv1.1 and TLSv1.2
+# SSLv23_method(), SSLv23_server_method(), SSLv23_client_method() are removed in 1.1.0
+
 when compileOption("dynlibOverride", "ssl"):
-  proc SSL_library_init*(): cint {.cdecl, dynlib: DLLSSLName, importc, discardable.}
-  proc SSL_load_error_strings*() {.cdecl, dynlib: DLLSSLName, importc.}
-  proc SSLv23_client_method*(): PSSL_METHOD {.cdecl, dynlib: DLLSSLName, importc.}
+  # Static linking
+  when defined(openssl_before_1_1):
+    proc SSL_library_init*(): cint {.cdecl, dynlib: DLLSSLName, importc, discardable.}
+    proc SSLv23_method*(): PSSL_METHOD {.cdecl, dynlib: DLLSSLName, importc.}
+    proc SSL_load_error_strings*() {.cdecl, dynlib: DLLSSLName, importc.}
+  else:
+    proc SSL_library_init*(): cint {.discardable.} =
+      ## Removed in OpenSSL 1.1.0
```

Comment:
Is SSL initialised implicitly in 1.1.0?

Reply:
@dom96 it seems that it's not, you need to use "OPENSSL_init_ssl"
https://wiki.openssl.org/index.php/Library_Initialization

Reply:
yeah, that's bad. Please make sure to test this under 1.1.0+

Reply:
I'm testing 8 combinations of static/dynamic linking, with/without the define, with older/newer libs installed. OPENSSL_init_ssl is initialized automatically, however manual init is still allowed so I added it in case someone wants to pass custom parameters.